File: src/com/designpatterns/creational/abstractfactory/RoundedRectangle.java
package com.designpatterns.creational.abstractfactory;

public class RoundedRectangle implements IShape {

    public static final String TAG = "RoundedRectangle";

    private final float length;
    private final float width;
    private final float roundedEdge;

    public RoundedRectangle(float length, float width, float roundedEdge) {
        this.length = length;
        this.width = width;
        this.roundedEdge = roundedEdge;
    }

    @Override
    public float area() {
        return length * width;
    }

    @Override
    public float perimeter() {
        return 2 * (length + width);
    }

    @Override
    public void draw() {
        System.out.printf(
                "Rectangle:: area:%f, perimeter:%f, roundedEdge:%f%n",
                area(), perimeter(), roundedEdge);
    }
}

File: src/com/designpatterns/creational/abstractfactory/kotlin/RoundedRectangle.kt
package com.designpatterns.creational.abstractfactory.kotlin

class RoundedRectangle(private val length: Float, private val width: Float, private val roundedEdge: Float) : IShape {

    override fun area(): Float {
        return length * width
    }

    override fun perimeter(): Float {
        return 2 * (length + width)
    }

    override fun draw() {
        System.out.printf("Rectangle:: area:%f, perimeter:%f, roundedEdge:%f%n", area(), perimeter(), roundedEdge)
    }

    companion object {
        const val TAG = "RoundedRectangle"
    }
}